src/engine-actions/prepare-swap.js

```diff
@@ -17,29 +8,38 @@ const SWAP_EXPIRY = '3600'
  * @type {String}
  * @default
  */
-const MEMO_PREFIX = 'sparkswap-swap-pivot:'
+const MEMO_PREFIX = 'sparkswap-swap-pivot'
 
 /**
  * Prepares for a swap in which this node is the counterparty to the intiating node
  *
- * @param {string} orderId - Order for which the swap is being executed
- * @param {string} swapHash        - swap hash that will be associated with the swap
- * @param {string} value   - Int64 string of the value of inbound currency
- * @returns {string} swapHash      Hash that will be translated
+ * @param {string} swapHash - swap hash that will be associated with the swap
+ * @param {string} value - Int64 string of the value of inbound currency
+ * @param {Date} expiryTime - absolute payment request expiry time
+ * @param {number} cltvExpiry - delta to use for the time-lock of the CLTV
+ *                              extended to the final hop (in seconds)
+ * @returns {string} paymentRequest - lightning invoice for a payment
  */
-async function prepareSwap (orderId, swapHash, value) {
+async function prepareSwap (swapHash, value, expiryTime, cltvExpiry) {
   this.logger.info(`Preparing swap for ${swapHash}`, { value })
+  // make prepareSwap idempotent by returning the existing invoice if one exists
+  try {
+    const { paymentRequest } = await lookupInvoice(swapHash, { client: this.client })
+    return paymentRequest
+  } catch (e) {
+    const expiry = Math.ceil((expiryTime - new Date()) / 1000)
```

Comment:
Seems like we would want to round this down, right? We want it to expire no later than the provided date?